### PR TITLE
add doc for using custom repo & custom entity defined in contenttype.yml

### PR DESCRIPTION
**Pull request by dadaxr**

### CODE REVIEW

File: docs/extensions/advanced/storage-repositories.md [hunk 1]
```diff
@@ -72,13 +72,14 @@ If your data are going to be stored in the `bolt_foo` table, you alias should be
 
 **Note:** the default table prefix is `bolt_`
 
-Custom repository to manage Entity defined in contenttype.yml
+Custom Repository to Manage Entity Defined in Contenttype.Yml
 -------------------------------------------------------------
-Sometimes it could be convenient to define content structure through the [contenttype.yml](../../contenttypes/intro)  file ( in particular to enjoy easy field declaration and to take advantage of the backend auto generated Create/Read/Update/Delete  UI ). If you also want to manage those content entities through a custom repository : we got you covered !
+Sometimes it could be convenient to define content structure through the [contenttype.yml](../../contenttypes/intro)  file ( in particular to enjoy easy field declaration and to take advantage of the backend auto generated Create/Read/Update/Delete  UI ). If you also want to manage those content entities through a custom repository: we got you covered !
```
Comment (PhillippOhlandt) [inline docs/extensions/advanced/storage-repositories.md]: You forgot one space here :)

**Comment by SahAssar** [comment]
Links look fine to me, they are in the relative format so they shouldn't lead to the crosslinking between bolt versions that we saw in the docs pre-rewrite.

File: docs/extensions/advanced/storage-repositories.md [hunk 2]
```diff
@@ -52,3 +52,173 @@ class KoalaCatcherExtension extends SimpleExtension
...
+```
+That alias should match the **"unprefixed"** table name your entity data should be stored in. 
+
+If your data are going to be stored in the `bolt_foo` table, you alias should be `foo`
+
+**Note:** the default table prefix is `bolt_`
+
+Custom Repository to Manage Entity Defined in Contenttype.Yml
+-------------------------------------------------------------
+Sometimes it could be convenient to define content structure through the [contenttype.yml](../../contenttypes/intro)  file ( in particular to enjoy easy field declaration and to take advantage of the backend auto generated Create/Read/Update/Delete  UI ). If you also want to manage those content entities through a custom repository: we got you covered!
```
Comment (rossriley) [inline docs/extensions/advanced/storage-repositories.md]: Sometimes it may be convenient

Comment (rossriley) [inline docs/extensions/advanced/storage-repositories.md]: change: If you also want to manage to: If you would like to manage

Comment (rossriley) [inline docs/extensions/advanced/storage-repositories.md]: change: we got you covered to: we've got you covered

File: docs/extensions/advanced/storage-repositories.md [hunk 3]
```diff
@@ -52,3 +52,173 @@ class KoalaCatcherExtension extends SimpleExtension
...
+
+Firstly You must create a custom Entity file which represent your content type and which extends `\Bolt\Storage\Entity\Content` and map that entity to your custom repository. Your custom repository must also extend `\Bolt\Storage\Repository\ContentRepository` and override the `createQueryBuilder` the whole is covered below:
```
Comment (rossriley) [inline docs/extensions/advanced/storage-repositories.md]: change to:

Firstly You must create a custom Entity file which represents your content type and which extends `\Bolt\Storage\Entity\Content` and then map that entity to your custom repository. Your custom repository must also extend `\Bolt\Storage\Repository\ContentRepository` and override the `createQueryBuilder` method as outlined below:

**Comment by GawainLynch** [comment]
Erm… that just went to master ;-)

**Comment by GawainLynch** [comment]
Also… just to flag … "ContentType" is our branding, not "content type"… can we keep an eye on this moving forward @PhillippOhlandt … I'll gt them to fixing up the branches for this one.

**Comment by PhillippOhlandt** [comment]
As said in slack, branch is correct. And sorry for the ContentType miss, didn't think about that while reading.

**Comment by GawainLynch** [comment]
Nothing to be sorry about. Flagging it helps us remind each other (we're human and prone to forget things). 

Also it keeps a public discussion visible, that gives our broader audience something to observe and learn from. :+1: 

I say… Team Work FTW!

**Pull request merged**